interfaces/builtin/multipass_support.go

```diff
@@ -0,0 +1,156 @@
+# configuring a TUN/TAP network on the host, with DHCP provided by a shared
+# "dnsmasq" process. File sharing between the VM and the host is provided by a
+# "sshfs_server" utility. All these utilities are shipped in the snap.
+
+# Each of these utilities have a very different different purpose, and an attempt
+# to confine them all in a single profile would result in an extremely broad AppArmor
+# profile.
+
+# Instead we defer to Multipass the responsibility of generating custom AppArmor
+# profiles for each of these utilities, and trust it launches each utility with
+# all possible security mechanisms enabled. The Multipass daemon itself will run
+# under this restricted policy.
+
+# Therefore this policy intentionally allows the Multipass daemon to configure
+# AppArmor.
+
+
+# Multipass has a server/client design, using a socket for IPC. The daemon runs
+# as root, but makes the socket accessible to anyone in the sudo group.
+# Need to permit chown and chgrp of the socket.
+capability chown,
+
+# Multipass uses gRPC for IPC, which calls shutdown on the socket
+unix ( shutdown) addr="@snap.@{SNAP_INSTANCE_NAME}.**",
+
+# Multipass generates AppArmor profiles for the utility processes it spawns.
+/sbin/apparmor_parser ixr,
+/etc/apparmor{,.d}/{,**} r,
+/etc/apparmor.d/{,cache/}multipass* rw,
+/sys/kernel/security/apparmor/{,**} r,
+/sys/kernel/security/apparmor/.remove w,
+/sys/kernel/security/apparmor/.replace w,
+
+# Allow running utility processes under the specialized AppArmor profiles.
+# These profiles will prevent utility processes escaping confinement.
+capability mac_admin,
+
+# Multipass will also use privilege separation when running utility processes
+capability setuid,
+capability setgid,
+
+# Some utility process (e.g. dnsmasq) will drop root privileges after startup and
+# change to another user. Multipass needs ability to stop them.
+capability kill,
+
+# Profiles Multipass generates have a naming scheme, restrict any profile changes to
+# those matching that scheme. Need unsafe to prevent env scrubbing.
+change_profile unsafe /var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/usr/bin/* -> multipass.*,
+signal (send) peer=multipass.*,
+ptrace (read, trace) peer=multipass.*,
+`
+
+const multipassSupportConnectedPlugSecComp = `
+# Description: allow operating as the Multipass daemon, and enable its various utilites
+# (qemu, qemu-img, dnsmasq, sshfs_server)
+
+# Note that this profile contains the union of all the necessary syscalls each of the
+# utilities requires. We rely on Multipass to generate specific AppArmor profiles
+# for each child process, to further restrict their abilities.
+
+# Multipass has server/client design, using a socket for IPC.
+# Note: Qemu has several features that also open sockets, but the AppArmor profile
+# Multipass will execute Qemu under locks down its filesystem access dramatically
+# and prevents it creating these sockets.
+accept
+accept4
+bind
+listen
```

Comment:
So, can you not use network-bind for this? (Did we discuss this before?)

Reply:
I can, this is my error, will be removed.